src/main/java/com/softserve/academy/spaced/repetition/controller/DeckCommentController.java

```diff
@@ -21,80 +21,61 @@
-        LOGGER.debug("View all comments for deck with id: {}", deckId);
-        List<Comment> commentsList = commentService.getAllCommentsForDeck(deckId);
-        Link collectionLink = linkTo(methodOn(DeckCommentController.class)
-                .getAllCommentsForDeck(categoryId, deckId)).withSelfRel();
-        List<CommentDTO> listOfComments = DTOBuilder
-                .buildDtoListForCollection(commentsList, CommentDTO.class, collectionLink);
-        List<CommentDTO> commentsTree = CommentDTO.buildCommentsTree(listOfComments);
-        return new ResponseEntity<>(commentsTree, HttpStatus.OK);
-    }
-
     @Auditable(action = AuditingAction.VIEW_COMMENT_FOR_DECK)
-    @GetMapping(value = "/api/category/{categoryId}/deck/{deckId}/comments/{commentId}")
-    public ResponseEntity<CommentDTO> getCommentById(@PathVariable Long categoryId,
-                                                     @PathVariable Long deckId,
-                                                     @PathVariable Long commentId) {
+    @GetMapping(value = "/{commentId}")
+    @ResponseStatus(HttpStatus.OK)
+    public CommentDTO getCommentById(@PathVariable Long deckId, @PathVariable Long commentId) {
         LOGGER.debug("View comment with id {} for deck with id: {}", commentId, deckId);
         DeckComment comment = commentService.getCommentById(commentId);
-        Link selfLink = linkTo(methodOn(DeckCommentController.class)
-                .getCommentById(categoryId, deckId, commentId)).withSelfRel();
-        CommentDTO commentDTO = DTOBuilder.buildDtoForEntity(comment, CommentDTO.class, selfLink);
-        return new ResponseEntity<>(commentDTO, HttpStatus.OK);
+        return buildDtoForEntity(comment, CommentDTO.class,
+                linkTo(methodOn(DeckCommentController.class)
+                        .getCommentById(deckId, commentId)).withSelfRel());
     }
 
     @Auditable(action = AuditingAction.CREATE_COMMENT_FOR_DECK)
-    @PostMapping(value = "/api/category/{categoryId}/deck/{deckId}/comment")
-    @PreAuthorize(value = "@accessToUrlService.hasAccessToDeck(#categoryId)")
-    public ResponseEntity<CommentDTO> addCommentForDeck(@Validated @RequestBody ReplyToCommentDTO replyToCommentDTO,
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public CommentDTO addCommentForDeck(@Validated @RequestBody ReplyToCommentDTO replyToCommentDTO,
                                                         @RequestBody String commentText,
-                                                        @PathVariable Long categoryId,
                                                         @PathVariable Long deckId) throws NotAuthorisedUserException {
         LOGGER.debug("Added comment to deck with id: {}", deckId);
         DeckComment comment = commentService
                 .addCommentForDeck(deckId, replyToCommentDTO.getCommentText(), replyToCommentDTO.getParentCommentId());
```

Comment:
I'm not sure, but maybe better to pass whole object `replyToCommentDTO`, not parts of it.

Reply:
I tried did like this, but it looks less readable in serviceImpl